src/type_checker/checker.rs

```diff
@@ -210,7 +210,7 @@ impl<B: Backend> TypeChecker<B> {
                     }
 
                     // `struct.field = <rhs>`
-                    ExprKind::FieldAccess { lhs, rhs } => {
```

Comment:
removal of variables shouldn't be in this PR as we want to make sure we're not forgetting to do something with them